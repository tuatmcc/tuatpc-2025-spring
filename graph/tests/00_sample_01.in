4 3
1 1 1
1 1
2
100
2 1
1 3
20 80
2 1
2 4
60 40
1 2
3
100
