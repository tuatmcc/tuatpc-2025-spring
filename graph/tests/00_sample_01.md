入力を図に表すと以下のようになります。
![graph1](https://www.dropbox.com/scl/fi/d88j2q2u15ihyf39pd49x/graph1.jpg?rlkey=xtilyyokldyo7qrrn0u8du1ky&st=5cwp48cl&raw=1)

楽器1から始めた場合、どのように二回遷移しても必ず音の種類が1の楽器になるため、求める確率は$1$になります。
楽器2から始めた場合、$1 1 1$の順で音を奏でるには
- 楽器2 → 楽器1 → 楽器2
- 楽器2 → 楽器3 → 楽器2


の順で遷移する必要があり、それぞれの確率は$0.2(0.2 × 1)$と$0.48(0.8 × 0.6)$です。したがって、求める確率は$0.68(0.2 + 0.48)$となります。

楽器3から始めた場合、$1 1 1$の順を音を奏でるには
- 楽器3 → 楽器2 → 楽器1
- 楽器3 → 楽器2 → 楽器3

の順で遷移する必要があり、それぞれの確率は$0.12(0.6 × 0.2)$と$0.48(0.6 × 0.8)$です。したがって、求める確率は$0.6(0.12 + 0.48)$となります。

楽器4から始めた場合、$1 1 1$の順で音を奏でることはできないため、求める確率は$0$になります。