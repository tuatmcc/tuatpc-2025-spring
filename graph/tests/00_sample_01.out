1 1
17 25
3 5
0 1
